### PR TITLE
XCOM-91 Add Oscar to the E-Commerce Project and add the Order Pipeline

### PR DESCRIPTION
**Pull request by stephensanchez**
This PR is for the following story about adding the E-Commerce Order Pipeline for Django-Oscar
https://openedx.atlassian.net/browse/XCOM-91

In addition, this installs django-oscar into the repository so oscar is made available from this project.
- Currently, all the URLs from django-oscar are made available. Another PR will need to be created in django-oscar-extensions to override the set of URLs we want to include from Oscar.
- The Authentication will have to be replaced with our own authentication system.
- There is an "image not found" JPEG included, but we may want to remove all the image attributes from our core Product.

The important aspect of this is the OSCAR_ORDER_STATUS_PIPELINE and the OSCAR_ORDER_STATUS_CASCADE.  These are defined based on the Order Flow below:

image

Please take a look when you get a chance: @dianakhuang @rlucioni @wedaly @rock345

### CODE REVIEW

File: requirements/base.txt [hunk 1]
```diff
@@ -7,3 +7,9 @@ django-model-utils==1.5.0   # BSD
 django-waffle==0.10         # BSD
 djangorestframework==3.0.5  # MIT
 logutils==0.3.3             # BSD
+
+# Our Django Oscar Requirements
+-e git+https://github.com/edx/django-oscar.git@1ce871a29b97789354b422ca559de956c6762aee#egg=django-oscar
+# Enable django-oscar-extensions once there is a setup.py and some actual extensions.
```
Comment (rlucioni) [inline requirements/base.txt]: Could you slap a `TODO` on here?

File: ecommerce/settings/base.py [hunk 2]
```diff
@@ -315,3 +328,49 @@
...
+    },
+}
+
+# TODO: Replace with new Authentication backend
+AUTHENTICATION_BACKENDS = (
+    'oscar.apps.customer.auth_backends.EmailBackend',
+    'django.contrib.auth.backends.ModelBackend',
+)
+
+########## END OSCAR SETTINGS
```
Comment (rlucioni) [inline ecommerce/settings/base.py]: Missing newline here; I'd also get rid of the newline above this line to stay consistent with the way sections are laid out in this file.

File: ecommerce/settings/base.py [hunk 3]
```diff
@@ -218,15 +230,16 @@
...
 # Apps specific for this project go here.
-LOCAL_APPS = (
+LOCAL_APPS = [
     'health',
-)
+]
 
 # See: https://docs.djangoproject.com/en/dev/ref/settings/#installed-apps
-INSTALLED_APPS = DJANGO_APPS + LOCAL_APPS
+from oscar import get_core_apps
```
Comment (rlucioni) [inline ecommerce/settings/base.py]: Can we move this import up to the top?

File: ecommerce/settings/base.py [hunk 4]
```diff
@@ -161,9 +168,11 @@
     'django.template.loaders.app_directories.Loader',
 )
 
+from oscar import OSCAR_MAIN_TEMPLATE_DIR
```
Comment (rlucioni) [inline ecommerce/settings/base.py]: Same here; can this be moved up to the top?

**Comment by rlucioni** [comment]
My nitpicks aside, this looks solid, :+1: from me.

File: ecommerce/settings/base.py [hunk 5]
```diff
@@ -315,3 +328,48 @@
 # Path of the SCSS file to use for the site's theme
 THEME_SCSS = 'sass/themes/open-edx.scss'
 ########## END THEME CONFIGURATION
+
+
+########## OSCAR SETTINGS
```
Comment (dianakhuang) [inline ecommerce/settings/base.py]: I guess one question I have is whether the Oscar settings should be in this repo? It seems like something as important as our statuses should be in django-oscar-extensions, etc, where we will be building out most of our oscar code?

Even if it shouldn't be in another repo, I think it makes sense to split these settings out into another file.

**Comment by dianakhuang** [comment]
Other than that, :ship:

**Pull request merged**